### PR TITLE
Feat: Fuzzy Log Search

### PR DESCRIPTION
**Pull request by Gathin23**
# Implementation

Added Fuzzy log search solving the following requirements:

- [x] Ignore case - A search for "error" should return logs with "... Error ... "
- [x] Char off-by-one - a search for "eror" should return logs with " ... error ... "

## Recording

Recording 2023-10-24 at 13 38 17

## Solves
#152

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -325,6 +325,7 @@
     "electron-debug": "^3.2.0",
     "electron-store": "^8.1.0",
     "electron-updater": "^6.1.5",
+    "fuse.js": "^6.6.2",
```
Comment (jgresham) [inline package.json:328]: extra?

Comment (Gathin23) [reply package.json:328]: Yes. The issue mentioned that I could use an npm package to solve it. So I used this for efficient working!

Comment (jgresham) [reply package.json:328]: I don't see it used (imported) in the code anywhere. Am I missing something?

Comment (Gathin23) [reply package.json:328]: Ahh.... Forgot to remove that package...i again found that the algorithm i coded worked so i replaced fuse with that...i forgot to update the PR and remove the package... Will do it now!!! 

Thanks

**Comment by Gathin23** [comment]
Updated: Removed the fuse package and used the logic which i tried intially, which works perfectly now. Now you can review the changes @jgresham

**Comment by jgresham** [comment]
> Updated: Removed the fuse package since I used my custom logic. Now you can review the changes @jgresham

+1 for not adding a dependency!

**Pull request merged**